diff -ru a/networking/httpd.c b/networking/httpd.c
--- a/networking/httpd.c	2013-03-16 17:26:23.000000000 -0400
+++ b/networking/httpd.c	2013-03-17 10:02:20.000000000 -0400
@@ -2174,7 +2174,18 @@
 		}
 	}
 	/* else fall through to send_file, it errors out if open fails: */
-
+	bb_error_msg("following through");
+#if ENABLE_FEATURE_HTTPD_GZIP
+	char *gzurl = xasprintf("%s.gz", tptr);
+	if (stat(tptr, &sb) != 0 && (content_gzip && stat(gzurl, &sb) != 0) && access("/cgi-bin/e404.cgi"+1, X_OK) == 0) {
+#else
+	if (stat(tptr, &sb) != 0 && access("/cgi-bin/e404.cgi"+1, X_OK) == 0) {
+#endif
+		bb_error_msg("getting through");
+		urlp[0] = '\0';
+		g_query = urlcopy;
+		send_cgi_and_exit("/cgi-bin/e404.cgi", prequest, length, cookie, content_type);
+	}
 	if (prequest != request_GET && prequest != request_HEAD) {
 		/* POST for files does not make sense */
 		send_headers_and_exit(HTTP_NOT_IMPLEMENTED);
Only in b/networking: .httpd.c.swp
